### PR TITLE
Replace nose2 with pytest

### PR DESCRIPTION
**Pull request by cheng-kevin**

### CODE REVIEW

File: run_benchmark_multi.sh [hunk 1]
```diff
@@ -16,15 +16,14 @@ hash=$(git rev-parse HEAD)
 make build-headless
 
 rm -rf "$work_dir/temp/"; mkdir "$work_dir/temp/"
-docker container stop garage_benchmark
-docker container rm garage_benchmark
+docker container stop garage_benchmark || true && docker rm garage_benchmark || true
```
Comment (ryanjulian) [inline run_benchmark_multi.sh]: i think this can be achieved by adding `--rm` to `docker run`

File: run_benchmark_multi.sh [hunk 2]
```diff
@@ -16,15 +16,14 @@ hash=$(git rev-parse HEAD)
...
 for FILE in "$work_dir/garage/tests/benchmarks/"*
     do
             testname=`basename $FILE .py`
             if [[ $testname == "test"* ]]
             then
                 echo "running test: $testname"
-                docker run --name garage_benchmark -e MJKEY="$(cat ~/.mujoco/mjkey.txt)" rlworkgroup/garage-headless nose2 -c setup.cfg tests.benchmarks.$testname
+                docker run --name garage_benchmark -e MJKEY="$(cat ~/.mujoco/mjkey.txt)" rlworkgroup/garage-headless pytest ./tests/benchmarks/$testname.py
```
Comment (ryanjulian) [inline run_benchmark_multi.sh]: hmm can you try using the garage makefile for this? it should be as simple as `make run-headless RUN_CMD="pytest tests/benchmarks/$testname.py"`

File: run_benchmark_multi.sh [hunk 3]
```diff
@@ -16,15 +16,14 @@ hash=$(git rev-parse HEAD)
 make build-headless
```
Comment (ryanjulian) [inline run_benchmark_multi.sh]: please make sure this script complied wit the [shell style guide](https://google.github.io/styleguide/shell.xml)

File: run_benchmark_multi.sh (hunk 1 above)
Comment (sud0nick) [reply run_benchmark_multi.sh]: @ryanjulian Would adding this cause the container to be deleted before the following `docker cp` command that copies the test results out?

Comment (ryanjulian) [reply run_benchmark_multi.sh]: Probably. The better way to do this is to use the garage Makefile. This redirects the data outputs of the container onto the host filesystem.

**Pull request closed without merging**